src/explorer/Queries/MultiColumn/ColumnPairs.cs

```diff
@@ -0,0 +1,46 @@
+﻿namespace Explorer.Queries
+{
+    using System.Collections.Generic;
+    using System.Collections.Immutable;
+    using System.Linq;
+    using System.Text.Json;
+
+    using Diffix;
+    using Explorer.Common;
+
+    public class ColumnPairs : DQuery<ColumnPairs.Result>
+    {
+        private ImmutableArray<string> Columns { get; set; } = ImmutableArray<string>.Empty;
+
+        public override Result ParseRow(ref Utf8JsonReader reader) => new Result(
+            ref reader, Columns);
+
+        protected override string GetQueryStatement(string table, IEnumerable<string> quotedColumns)
+        {
+            // Unquote the columns.
+            Columns = quotedColumns.Select(c => c[1..^1]).ToImmutableArray();
```

Comment:
I disagree tbh. This will only push the same code a level higher in the call chain and duplicates the data that needs to be passed.

Reply:
If you look at `DQueryStatement.BuildQueryStatement`, there you receive the unquoted columns and then the columns are quoted, and passed to `GetQueryStatement`. It is simple to modify that place to call `GetQueryStatement` for multi-column queries with both sets of names, without requiring the unquoting operation.

Another possible solution to remove the unquoting (and arguably better), will be to pass the `groupingLabels` to the `ColumnPairs` constructor from the place where the query is used.

Reply:
Since the data is the same, just with a different representation (quoted vs. unquoted), I think it makes most sense for this to be explicit in the code, and for the different representations to be derived from the same source. 

Otherwise it's possible for example to add labels that don't match the columns (or at least, from the perspective of someone reading the method signature, it suggests that the labels and columns are independent, which isn't the case). 

Also, it's not certain that all multi-column queries will require a list of columns and a list of quoted columns - so changing the method signature requires adapting the interface's method signature, meaning all multi-column queries would need to take the unquoted column list, even if they don't need it. 

IMO the cleanest solution is to take the (unquoted) column list in the constructor and use this construct the query (ie. add the quotes). But this would also break the interface's method signature. 

The current solution is not perfect, I agree. But it avoids any heavy changes to the interface while making sure that the labels match requested column names...